### PR TITLE
Lumin Platform Support

### PR DESCRIPTION
**Pull request by StephenHodgson**
Added Lumin Support to the XRTK.

- MagicLeap SDK 0.20.0 install required and not included.
  - Future work aims to remove this dependency by getting data directly from native dlls

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,9 +1,12 @@
+[submodule "Submodules/UpmGitExtension"]
+	path = Submodules/UpmGitExtension
+	url = https://github.com/XRTK/UpmGitExtension.git
 [submodule "Submodules/WindowsMixedReality"]
 	path = Submodules/WindowsMixedReality
 	url = https://github.com/XRTK/WindowsMixedReality.git
 [submodule "Submodules/SDK"]
 	path = Submodules/SDK
 	url = https://github.com/XRTK/SDK.git
-[submodule "Submodules/UpmGitExtension"]
```
Comment (StephenHodgson) [inline .gitmodules]: Need to revert this change

File: XRTK-Core/Assets/SymbolicLinkSettings.asset [hunk 2]
```diff
@@ -13,12 +13,15 @@ MonoBehaviour:
   m_Name: SymbolicLinkSettings
   m_EditorClassIdentifier: 
   symbolicLinks:
+  - sourceRelativePath: /Submodules/UpmGitExtension/XRTK.UpmGitExtension/Assets/XRTK.UpmGitExtension
```
Comment (StephenHodgson) [inline XRTK-Core/Assets/SymbolicLinkSettings.asset]: revert this change

File: XRTK-Core/Assets/XRTK/Definitions/Devices/ControllerMappingLibrary.cs [hunk 3]
```diff
@@ -280,7 +280,8 @@ private static Texture2D GetControllerTextureInternal(SupportedControllerType co
             {
                 foreach (var profile in profiles.ControllerMappingProfiles)
                 {
-                    if (profile.ControllerType == controllerType)
+                    if (profile != null &&
```
Comment (StephenHodgson) [inline XRTK-Core/Assets/XRTK/Definitions/Devices/ControllerMappingLibrary.cs]: pull out this change

File: XRTK-Core/Assets/XRTK/Inspectors/Profiles/MixedRealityControllerDataProviderProfileInspector.cs [hunk 4]
```diff
@@ -15,6 +15,7 @@ public class MixedRealityControllerDataProviderProfileInspector : BaseMixedReali
     {
         private static readonly GUIContent AddControllerDataProviderContent = new GUIContent("+ Add a New Controller Data Provider");
         private static readonly GUIContent RemoveControllerDataProviderContent = new GUIContent("-", "Remove Controller Data Provider");
+        private static readonly GUIContent ProfileContent = new GUIContent("Profile");
```
Comment (StephenHodgson) [inline XRTK-Core/Assets/XRTK/Inspectors/Profiles/MixedRealityControllerDataProviderProfileInspector.cs]: pull out this change

File: XRTK-Core/Assets/XRTK/Inspectors/PropertyDrawers/TypeReferencePropertyDrawer.cs [hunk 5]
```diff
@@ -248,7 +247,6 @@ private static void DrawTypeSelectionControl(Rect position, SerializedProperty p
                         if (!reference.Contains(" {missing}"))
                         {
                             reference += " {missing}";
-                            Debug.LogError($"Failed to resolve class reference for property '{property.name}' on {property.serializedObject.targetObject.name}");
```
Comment (StephenHodgson) [inline XRTK-Core/Assets/XRTK/Inspectors/PropertyDrawers/TypeReferencePropertyDrawer.cs]: pull out these changes

File: XRTK-Core/Assets/XRTK/Services/MixedRealityToolkit.cs [hunk 6]
```diff
@@ -430,12 +430,6 @@ private void InitializeServiceLocator()
                     {
                         foreach (var spatialObserver in ActiveProfile.SpatialAwarenessProfile.RegisteredSpatialObserverDataProviders)
                         {
-                            if (spatialObserver.Profile == null)
-                            {
-                                Debug.LogError($"Missing profile for {spatialObserver.SpatialObserverName}");
```
Comment (StephenHodgson) [inline XRTK-Core/Assets/XRTK/Services/MixedRealityToolkit.cs]: pull out this change

File: XRTK-Core/ProjectSettings/ProjectVersion.txt [hunk 7]
```diff
@@ -1 +1,2 @@
-m_EditorVersion: 2018.3.13f1
+m_EditorVersion: 2019.1.0f2
```
Comment (StephenHodgson) [inline XRTK-Core/ProjectSettings/ProjectVersion.txt]: pull out this change

**Comment by SimonDarksideJ** [comment]
Shouldn't this be targetting the Lumin Extension Project ?

**Comment by StephenHodgson** [comment]
They do. Some of those changes are in the submodule. Some of the changes need to happen in this project as well.
